src/web/app/pages-session/session-submission-page/session-submission-page.component.ts

```diff
@@ -555,4 +593,128 @@ export class SessionSubmissionPageComponent implements OnInit {
       this.statusMessageService.showErrorMessage(resp.error.message);
     });
   }
+
+  /**
+   * Creates a request to change a comment.
+   * The request can be either of the following: DELETE, CREATE or UPDATE.
+   */
+  private createCommentChangeRequest(questionSubmissionFormModel: QuestionSubmissionFormModel,
+      recipientSubmissionFormModel: FeedbackResponseRecipientSubmissionFormModel): Observable<any> {
+
+    if (!recipientSubmissionFormModel.comment) {
+      return of({});
+    }
+
+    // Create delete comment request if it has an empty comment text.
+    if (recipientSubmissionFormModel.comment.commentText === '') {
+      return this.commentService.deleteComment(recipientSubmissionFormModel.comment.commentId.toString());
+    }
+
+    // If existing comment, create update request.
+    if (recipientSubmissionFormModel.comment.commentId) {
```

Comment:
Will this be correct? since the default value for `commentId` is `-1` and `if (-1) {...}` will excute codes inside. Please verify this.

Reply:
In this case the default value never gets used because the comment is set to undefined by the parent if its nonexistent. But you're right, changed it to `NaN` instead to be safer.